php/commands/sites.php

```diff
@@ -228,15 +229,28 @@ public function aliases($args, $assoc_args) {
     $print = Input::optional('print', $assoc_args, false);
     $json = \Terminus::get_config('json');
     $location = Input::optional('location', $assoc_args, getenv("HOME").'/.drush/pantheon.aliases.drushrc.php');
-    $message = "Pantheon aliases updated.";
-    if (!file_exists($location)) {
-      $message = "Pantheon aliases created.";
+
+    // Cannot provide just a directory
+    if (is_dir($location)) {
+      \Terminus::error("Please provide a full path with filename, e.g. %s/pantheon.aliases.drushrc.php", $location);
+      exit(1);
+    }
+
+    $file_exists = file_exists($location);
+
+    // Create the directory if it doesn't yet exist
+    $dirname = dirname($location);
+    if (!is_dir($dirname)) {
+      mkdir($dirname, 0755, true);
     }
+
     $content = $user->getAliases();
     $h = fopen($location, 'w+');
     fwrite($h, $content);
     fclose($h);
-    chmod($location, 0777);
+    chmod($location, 0755);
```

Comment:
The typical use-case for Drush is to call it from only one user account -- or, to say it another way, every user who wants to use Drush installs their own copy, generally via `composer global require drush/drush`.

Given that this is the case, `0700` is sufficient permissions for this file to be accessed in typical usage.  There are reasons why you might not want `0755` or `0750` e.g. when using Drush in a multi-user environment.  This should be rare for Terminus users, since typically the local machine will not be shared, and the remote machine will be Pantheon.  In the balance, though, `0700` is probably the best choice, if you had to choose something.

The other option, though, is you could simply not chmod it, and trust that the user has set their umask appropriately for their situation.  The umask shouldn't give you less then `0700`, so that should be safe too.